server/src/services/QwixxLogic.ts

```diff
@@ -47,4 +47,24 @@ export default class QwixxLogic {
   get players() {
     return this._playersArray;
   }
+
+  serialize() {
+    const serializedPlayers = this._playersArray.reduce((acc, player) => {
+      acc[player.name] = player.serialize();
+      return acc;
+    }, {} as Record<string, any>);
```

Comment:
I'm not very good with ".reduce()". But I guess it adds the "player.serialize()" value to a "player.name" key and then it collects all of them in an object. Then below it assigns it to a "players" key right?

Reply:
The reduce method is used to create a players object which then has key value pairs of players name and then the gamecard object. This was done to match the current data structure of the front end. As currently the backend has a playersArray which contains the player name and associated gamecard. Not sure if it would make sense to make the data structure match on the backend.